Services/WindowServer/WindowManager.cpp

```diff
@@ -379,6 +379,9 @@ void WindowManager::notify_rect_changed(Window& window, const Gfx::IntRect& old_
 
     tell_wm_listeners_window_rect_changed(window);
 
+    if (window.type() == WindowType::MenuApplet)
+        MenuManager::the().set_needs_window_resize();
```

Comment:
We can and originally did! Thought I was missing some logic because of strange text artifacting, but it seems a separate issue, and I forgot to change it back. Dunno if KVM=disabled exaggerates the problem, but the applets in menubar have never drawn correctly for me on boot. Requires an additional action like focus change to update.